### ISSUE TITLE
[BUG] When a variable references a property of a resource it errors with `error parsing Bicep variables`

### ISSUE THREAD

**Issue by TheGrussalo** [christosgalano/bicep-docs#38]
When a variable is referencing a property from a resource (an App Service Plan) it errors when creating the document.

## Description
When running the command:
```pwsh
bicep-docs.exe -V --input . --output README.md
```
The error is returned:
```
error processing main.bicep: error parsing Bicep variables
```

## Steps to Reproduce
- Create a `main.go` file with the following contents:
```bicep
@description('Name of the App Service Plan to host Web-App on')
param servicePlanName string

@description('Get resiliance options from Service Plan')
var isZoneRedundant = servicePlan.properties.zoneRedundant
var reserved = servicePlan.properties.reserved


@description('Get App Service Plan Object')
resource servicePlan 'Microsoft.Web/serverfarms@2023-01-01' existing =  {
   name: servicePlanName 
}

output isZoneRedundant bool = isZoneRedundant
output reserved bool = reserved
```

## Context
I'm trying to use the tool to iterate through some in-house custom modules and there are a couple of failures.
This is one of our older modules which may be re-written but thought it was worth raising a bug.
Thanks for fixing the other defect so quickly! :)

## Your Environment

- Version used: bicep-docs version v1.2.1
- Operating System and version: Windows 11 23H2

**Comment by christosgalano** [comment]
First of all thanks for your feedback once again @TheGrussalo, it is crucial in improving the tool since I cannot test/think of every possible scenario.

This is expected behavior due to how Bicep compiles variables that reference resource properties. Let's look at the process:

Original Bicep:
```bicep
var isZoneRedundant = servicePlan.properties.zoneRedundant
var reserved = servicePlan.properties.reserved

resource servicePlan 'Microsoft.Web/serverfarms@2023-01-01' existing = {
    name: servicePlanName
}

output isZoneRedundant bool = isZoneRedundant
output reserved bool = reserved
```

Compiled ARM template:
```json
{
  "parameters": {
    "servicePlanName": {
      "type": "string",
      "metadata": {
        "description": "Name of the App Service Plan to host Web-App on"
      }
    }
  },
  "resources": [],
  "outputs": {
    "isZoneRedundant": {
      "type": "bool",
      "value": "[reference(resourceId('Microsoft.Web/serverfarms', parameters('servicePlanName')), '2023-01-01').zoneRedundant]"
    },
    "reserved": {
      "type": "bool",
      "value": "[reference(resourceId('Microsoft.Web/serverfarms', parameters('servicePlanName')), '2023-01-01').reserved]"
    }
  }
}
```

During compilation, Bicep optimizes the template by:
1. Evaluating variable references
2. Converting them directly into `reference()` function calls
3. Omitting the variables section from the ARM template

This optimization is logical and doesn't affect the template's functionality - the result is the same whether the properties are referenced through variables or directly. The variables provide a more convenient way to work with these values in Bicep.

Since bicep-docs parses the produced ARM template to generate documentation, it cannot find the variables section, leading to the error you're seeing.

**Comment by christosgalano** [comment]
First of all thanks for your feedback once again @TheGrussalo, it is crucial in improving the tool since I cannot test/think of every possible scenario.

This is expected behavior due to how Bicep compiles variables that reference resource properties. Let's look at the process:

Original Bicep:
```bicep
var isZoneRedundant = servicePlan.properties.zoneRedundant
var reserved = servicePlan.properties.reserved

resource servicePlan 'Microsoft.Web/serverfarms@2023-01-01' existing = {
    name: servicePlanName
}

output isZoneRedundant bool = isZoneRedundant
output reserved bool = reserved
```

Compiled ARM template:
```json
{
  "parameters": {
    "servicePlanName": {
      "type": "string",
      "metadata": {
        "description": "Name of the App Service Plan to host Web-App on"
      }
    }
  },
  "resources": [],
  "outputs": {
    "isZoneRedundant": {
      "type": "bool",
      "value": "[reference(resourceId('Microsoft.Web/serverfarms', parameters('servicePlanName')), '2023-01-01').zoneRedundant]"
    },
    "reserved": {
      "type": "bool",
      "value": "[reference(resourceId('Microsoft.Web/serverfarms', parameters('servicePlanName')), '2023-01-01').reserved]"
    }
  }
}
```

During compilation, Bicep optimizes the template by:
1. Evaluating variable references
2. Converting them directly into `reference()` function calls
3. Omitting the variables section from the ARM template

This optimization is logical and doesn't affect the template's functionality - the result is the same whether the properties are referenced through variables or directly. The variables provide a more convenient way to work with these values in Bicep.

The tool previously failed because it was expecting a 1:1 match between variables found in the Bicep file and those in the ARM template. When Bicep optimized away the variables section, this check failed, resulting in the error message.

This has been fixed in version 1.2.2. The tool now:
1. Successfully parses variables from the Bicep file
2. Recognizes that they might be optimized away in the ARM template
3. Uses the Bicep variables directly instead of requiring them to exist in the ARM template

You can now use variables that reference resource properties without any errors from bicep-docs.

**Comment by TheGrussalo** [comment]
Thank you once again, works perfectly!
